openedx/features/marketplace/views.py

```diff
@@ -1,24 +1,145 @@
+    ordering = ['-created']
+    template_engine = 'mako'
 
-    def get(self, request, *args, **kwargs):
-        return render_to_response(self.template_name, {})
+    def get_checked_choices(self, choices):
+        """
+        This method takes list of tuples (choices) and returns a list
+        of choice keys that exist in self.request.GET
+        :param: choices
+        :return: list of choice keys that are also in self.request.GET
+        """
+        params = self.request.GET
+        tuple_key_index = 0
+        return [choice[tuple_key_index] for choice in choices if params.get(choice[tuple_key_index])]
+
+    def get_queryset(self):
+        params = self.request.GET
+
+        query = params.get(MARKETPLACE_PARAM_QUERY_KEY)
+        city = params.get(MARKETPLACE_PARAM_CITY_KEY)
+        country = params.get(MARKETPLACE_PARAM_COUNTRY_KEY)
+
+        request_type_list = self.get_checked_choices(ORGANIZATIONAL_PROBLEM_CHOICES_VIEW)
+        organizational_sector_list = self.get_checked_choices(ORGANIZATION_SECTOR_CHOICES_VIEW)
+        queryset = super(MarketplaceListView, self).get_queryset()
+
+        if request_type_list:
+            # has other problem selected
+            if ORGANIZATIONAL_PROBLEM_TYPE_OTHER_KEY in request_type_list:
+                queryset = queryset.filter(reduce(operator.or_, (~Q(organizational_problems__endswith='|')
+                                                                 | Q(organizational_problems__contains=x)
+                                                                 for x in request_type_list)))
+            else:
+                queryset = queryset.filter(reduce(operator.or_, (Q(organizational_problems__contains=x)
+                                                                 for x in request_type_list)))
+        if organizational_sector_list:
+            # has other sector selected
+            if ORGANIZATIONAL_SECTOR_OTHER_KEY in organizational_sector_list:
+                queryset = queryset.filter(reduce(operator.or_, (~Q(organization_sector__endswith='|')
+                                                                 | Q(organization_sector__contains=x)
+                                                                 for x in organizational_sector_list)))
+            else:
+                queryset = queryset.filter(reduce(operator.or_, (Q(organization_sector__contains=x)
+                                                                 for x in organizational_sector_list)))
+
+        if country:
+            country_codes = [django_country[DJANGO_COUNTRIES_KEY_INDEX] for django_country in countries if
+                             country.lower() in django_country[DJANGO_COUNTRIES_VALUE_INDEX].lower()]
+            queryset = queryset.filter(country__in=country_codes)
+        if city:
+            queryset = queryset.filter(city__icontains=city)
+        if query:
+            queryset = queryset.filter(organization_mission__icontains=query)
```

Comment:
Are we sure that the query is only to be searched on the mission and not the org name? Or both? Because I see no way in which the user can search for all requests from a specific organization.

Reply:
Use appropriate name for this queryset and query that defines their purpose.